packages/codemirror-extension/src/commands.ts

```diff
@@ -90,6 +90,27 @@ export const commandsPlugin: JupyterFrontEndPlugin<void> = {
       return !!findEditorView();
     };
 
+    const isCodeFoldEnabled = () => {
+      const currentWidget = app.shell.currentWidget;
+      // Check for active notebook
+      if (
+        tracker.currentWidget !== null &&
+        tracker.currentWidget === currentWidget
+      ) {
+        return true;
+      }
+      // Check for file editor with focus method
+      const fileEditorWidget = currentWidget as any;
+      if (
+        fileEditorWidget &&
+        fileEditorWidget.content &&
+        typeof fileEditorWidget.content.editor?.focus === 'function'
+      ) {
+        return true;
+      }
```

Comment:
Can you extract this logic to a helper function and reuse it in both `execute` of individual commands and in this function? This is to ensure that they stay in sync.